src/gurt/fault_inject.c

```diff
@@ -323,8 +323,9 @@ one_fault_attr_parse(yaml_parser_t *parser)
 			attr.fa_max_faults = val;
 			D_DEBUG(DB_ALL, "max_faults: %lu\n", val);
 		} else if (!strcmp(key_str, err_code)) {
-			attr.fa_err_code = val;
-			D_DEBUG(DB_ALL, "err_code: %lu\n", val);
+			attr.fa_err_code = strtol(val_str, NULL, 0);
```

Comment:
any reason why not change line 308 instead, and add error check to ensure val >=0 for strcmp cases when value is not expected to be negative?

Reply:
Because the other types are all unsigned?

I think I put in a commit message but the right approach would be to use fscanf for everything, but I think that's overkill for developer-only functionality.

Reply:
they are unsigned, but if someone accidentally specifies negative value for those fields in config file, they will end up being converted to 0 by strtoul. if we always did strtol and checked the ones that supposed to be unsigned to be >=0 that would make it easier to identify potential problems in config file.
Anyway, not a big issue for me